### PR TITLE
feat(security): HMAC cookie auth for webchat — eliminate build-time API key exposure

### LINKED ISSUE #623: feat(security): webchat HMAC cookie authentication — eliminate build-time API key exposure
## Problem

Current webchat API Key flows through `.env.local` → `next.config.mjs` → JS bundle → `go:embed`, resulting in:

1. **Binary leaks secrets**: Published hotplex binary contains plaintext API Key
2. **Deployment binding**: Different deployments require webchat rebuild
3. **OWASP violation**: API Key in JS is readable via XSS

## Solution

HMAC-SHA256 signed cookies (code-server verification pattern):

- Gateway generates HMAC key at startup via `crypto/rand` — binary contains zero secrets
- First webchat visit auto-issues `HttpOnly` cookie
- Subsequent REST/WebSocket requests authenticate via cookie, no frontend API Key needed

## Spec

Full design: `docs/superpowers/specs/2026-06-02-webchat-cookie-auth-design.md`

### Scope

| Area | Files | Change |
|------|-------|--------|
| Go: cookie module | `internal/security/cookie.go` (new) | `CookieAuth` — HMAC-SHA256 sign/verify |
| Go: auth fallback | `internal/security/auth.go` | 3rd priority cookie in `AuthenticateRequest` |
| Go: WS upgrade | `internal/gateway/hub.go` | Cookie auth path in `HandleHTTP` |
| Go: SPA handler | `internal/webchat/server.go` | Auto-issue cookie on index.html |
| Go: DI wiring | `cmd/hotplex/gateway_run.go` | Create + inject `CookieAuth` |
| TS: config | `webchat/lib/config.ts` | Remove build-time `apiKey`, add `isSameOrigin()` |
| TS: HTTP calls | `webchat/lib/api/sessions.ts` | `credentials: same-origin` instead of `X-API-Key` |
| TS: WS adapter | `webchat/lib/adapters/hotplex-runtime-adapter.ts` | Omit `authToken` in same-origin |
| Build | `webchat/next.config.mjs` | Clean up `HOTPLEX_WEBCHAT_API_KEY` forwarding |
| Tests | `internal/security/cookie_test.go` (new) | 7 test cases from spec §9 |

### Backward Compatibility

- `X-API-Key` header auth **fully preserved** (priority 1)
- `api_key` query param auth **fully preserved** (priority 2)
- Cookie is **3rd priority fallback**, only when header + query both empty
- External webchat deployments (non-same-origin) continue using header auth

### Security

| Threat | Mitigation |
|--------|-----------|
| Binary reverse engineering | HMAC key generated at runtime, not in binary |
| XSS credential theft | `HttpOnly` — JS cannot read cookie |
| CSRF | `SameSite=Strict` |
| Cookie forgery | HMAC-SHA256 signature verification |
| Replay | Built-in timestamp, 24h expiry |

### PR DESCRIPTION
**Pull request by aaronwong1989**
## Problem

Closes #623

当前 webchat 的 API Key 通过 `.env.local` → `next.config.mjs` → JS bundle → `go:embed` 链路硬编码到二进制中，导致：

1. **二进制泄露密钥**：发布的 hotplex 二进制包含明文 API Key
2. **部署绑定**：不同部署需要重新构建 webchat
3. **违反 OWASP**：API Key 存储在 JS 中可被 XSS 读取

## Solution

HMAC-SHA256 签名 Cookie 认证（运行时生成密钥，零 secret 嵌入）：

```
浏览器 GET / → 检测无 cookie → 签发 HttpOnly cookie → 返回 index.html
后续请求 → 浏览器自动携带 cookie → Gateway 验证 HMAC 签名
```

### Go Backend

| File | Change |
|------|--------|
| `internal/security/cookie.go` | **New**: `CookieAuth` — HMAC-SHA256 签发/验证，`crypto/rand` 密钥 |
| `internal/security/auth.go` | `AuthenticateRequest` 增加 cookie 第三优先级回退 |
| `internal/gateway/hub.go` | `HandleHTTP` WS 升级时检查 cookie |
| `internal/webchat/server.go` | SPA fallback 自动签发 cookie |
| `cmd/hotplex/gateway_run.go` | DI 创建 `CookieAuth` 并注入 |

### TypeScript Frontend

| File | Change |
|------|--------|
| `webchat/lib/config.ts` | `apiKey` 默认空，新增 `isSameOrigin()` |
| `webchat/lib/api/sessions.ts` | 同源 → `credentials:same-origin`，跨源 → `X-API-Key` |
| `webchat/lib/adapters/hotplex-runtime-adapter.ts` | 同源省略 `authToken` |

### Tests

8 个新测试：签名验证 round-trip、过期拒绝、篡改拒绝、Secure flag、不重复签发、cookie+botID。

## Backward Compatibility

- `X-API-Key` header 认证 **完全保留**（优先级 1）
- `api_key` query param 认证 **完全保留**（优先级 2）
- Cookie 是第三优先级回退
- 外部部署 webchat（非同源）继续使用 header 认证

## Security

| 威胁 | 防护 |
|------|------|
| 二进制泄露 | HMAC 密钥运行时生成 |
| XSS 窃取 | `HttpOnly` 不可读 |
| CSRF | `SameSite=Strict` |
| 伪造 | HMAC-SHA256 签名验证 |
| 重放 | 内置 timestamp，24h 过期 |

### CODE REVIEW

File: internal/gateway/hub_test.go [hunk 1]
```diff
@@ -1041,7 +1041,7 @@ func TestHub_HandleHTTP_Success(t *testing.T) {
 	handler := NewHandler(HandlerDeps{Log: slog.Default(), Hub: h})
```
Comment (hotplex-ai) [inline internal/gateway/hub_test.go:1041]: **[P2]** 所有 5 个 HandleHTTP 测试均传 `nil` 作为 cookieAuth 参数，cookie 认证的 WS 升级路径未被覆盖。建议新增 `TestHub_HandleHTTP_CookieAuth` 测试用例，验证：
1. 携带有效 cookie 的 WS 连接成功认证
2. 携带无效/过期 cookie 的连接回退到 pendingAuth

File: webchat/lib/api/sessions.ts [hunk 2]
```diff
@@ -5,10 +5,26 @@
  * using X-API-Key header for authentication.
```
Comment (hotplex-ai) [inline webchat/lib/api/sessions.ts]: **[P3]** 文件头注释 "using X-API-Key header for authentication" 已不准确——同源模式下使用 `credentials: 'same-origin'` cookie 认证，不再发送 X-API-Key header。建议更新为描述双模式认证策略。

**Review by hotplex-ai — DISMISSED** [review]
## Code Review — hotplex-ai

**Verdict: APPROVE** | P0:0 P1:0 P2:1 P3:2

### Summary

HMAC cookie auth 实现质量高：crypto/rand 密钥生成、HMAC-SHA256 签名验证（constant-time compare）、HttpOnly + SameSite=Strict + conditional Secure、24h 过期、同源自动签发不重复刷新。向后兼容性完整（X-API-Key / query param 优先级不变），前端同源/跨源条件认证逻辑清晰。

### P2 Findings

1. **hub_test.go 缺少 cookie auth 路径集成测试** — 所有 5 个 HandleHTTP 测试均传 `nil` cookieAuth，WS 升级的 cookie 认证分支未被测试覆盖。建议新增 `TestHub_HandleHTTP_CookieAuth`。

### P3 Findings

2. **cookieAuth 双路径注入** — 同一个 `*CookieAuth` 指针同时存入 `Authenticator.cookieAuth`（用于 HTTP API）和 `GatewayDeps.CookieAuth`（用于 WS HandleHTTP 参数），未来维护可能偏离。考虑统一通过 Authenticator 访问。

3. **sessions.ts 文件头注释过时** — `webchat/lib/api/sessions.ts:5` 仍写 "using X-API-Key header for authentication"，同源模式已改用 cookie credentials，注释应更新。

### Verified OK

- HMAC 签名 raw bytes 含 `|` (0x7C) 不影响 verify — `SplitN(_, "|", 3)` 停在 2 次分割，第三部分完整保留 HMAC 字节 ✅
- RWMutex 使用正确 — SetCookieAuth 用 Lock 写，AuthenticateRequest 用 RLock 读 ✅
- 无 double unlock — 手动 unlock 模式与既有代码一致 ✅
- 向后兼容 — apiKey 默认值从 "dev" 改为 "" 不影响跨源部署（有 HOTPLEX_WEBCHAT_API_KEY 时仍传 header）✅

File: internal/security/cookie.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+// sign creates a signed cookie value: Base64(timestamp|userID|hmac).
+func (c *CookieAuth) sign(userID string) string {
+	ts := time.Now().Unix()
+	payload := fmt.Sprintf("%d|%s", ts, userID)
+
+	mac := hmac.New(sha256.New, c.secret)
+	mac.Write([]byte(payload))
+	sig := mac.Sum(nil)
+
+	return base64.RawURLEncoding.EncodeToString([]byte(payload + "|" + string(sig)))
```
Comment (hotplex-ai) [inline internal/security/cookie.go]: **[P1] HMAC 签名拼接 correctness bug** — `mac.Sum(nil)` 返回 32 字节原始二进制，其中 `|`(0x7C) 出现概率约 12%。`string(sig)` 拼入 `|` 分隔字符串后，`verify()` 中 `SplitN(raw, "|", 3)` 会错误截断签名。建议对签名做 hex 编码：`fmt.Sprintf("%d|%s|%x", ts, userID, sig)`

File: internal/webchat/server.go [hunk 4]
```diff
@@ -68,6 +71,10 @@ func Handler(csp string) http.Handler {
 		}
 
 		// SPA fallback: serve index.html for all non-file paths.
+		// Issue a cookie if cookieAuth is configured and request lacks a valid one.
+		if cookieAuth != nil {
+			_ = cookieAuth.SetCookie(w, r, "webchat_user")
```
Comment (hotplex-ai) [inline internal/webchat/server.go:80]: **[P1] 硬编码 `"webchat_user"`** 使所有 webchat 访客共享同一身份。REST API session 所有权检查 (`si.UserID != userID`) 下，cookie 用户无法访问通过 API key 创建的 session。如果是初始版本已知限制，建议添加 `// TODO(security): support real user identity` 注释。

**Review by hotplex-ai — COMMENTED** [review]
## Code Review — hotplex-ai

Verdict: ✅ APPROVE (with suggestions) | P0:0 P1:2 P2:3 P3:1

---

### P1 Findings

**1. HMAC 签名原始二进制拼入 `|` 分隔字符串** (`internal/security/cookie.go:90-92`)

`sign()` 中 `mac.Sum(nil)` 返回原始 32 字节，通过 `string(sig)` 拼入 `|` 分隔的字符串。HMAC 输出中 `|`(0x7C) 字节出现概率约 12%，会导致 `verify()` 中 `SplitN(raw, "|", 3)` 截断签名，使合法 cookie 被拒绝。

建议：对签名做 hex 编码后再拼接：
```go
// sign
return base64.RawURLEncoding.EncodeToString([]byte(
    fmt.Sprintf("%d|%s|%x", ts, userID, sig),
))
// verify 中对应 hex.DecodeString(sigStr)
```

**2. 硬编码 `"webchat_user"` 作为所有 cookie 用户身份** (`internal/webchat/server.go:76`)

所有 webchat 访客共享同一 `"webchat_user"` 身份。当前为单用户 webchat 场景设计，但 session REST API 有所有权检查 (`si.UserID != userID`)。如果是已知初始版本限制，建议添加 TODO 注释说明后续支持真实用户身份。

### P2 Findings

**3. `time.Sleep` 违反项目测试规范** (`internal/security/cookie_test.go:68`)

`TestCookieAuthExpiry` 中 `time.Sleep(10 * time.Millisecond)` 违反 CLAUDE.md 禁止 `time.Sleep` 等待的规范。`maxAge` 已设为 `1ns`，无需等待——直接验证即可移除 sleep。

**4. 双路径 cookie auth 无文档约束** (`hub.go:392-398` + `auth.go:68-74`)

`HandleHTTP` 直接调用 `cookieAuth.Authenticate()` (WS 升级)，`AuthenticateRequest` 通过 `a.mu.RLock()` 保护调用 (REST API)。两条路径绕过 Authenticator 锁。当前安全（启动时设置一次后不变），但缺少 immutability 约束注释，未来密钥轮换可能引入竞态。

**5. `X-Forwarded-Proto` 无条件信任** (`internal/security/cookie.go:137`)

Gateway 直连时攻击者可设置 `X-Forwarded-Proto: https`，导致 cookie 被标记 Secure 但实际通过 HTTP 传输（浏览器不会回传→功能受损）。属于 self-DoS 而非安全绕过，但建议增加配置开关控制是否信任 forwarded headers。

### P3 Finding

**6. cookie.go 包级注释误导 godoc** (`internal/security/cookie.go:1-4`)

文件头 `// Package security provides HMAC-SHA256 signed cookie authentication...` 覆盖了 `auth.go` 等其他文件的 package doc。建议改为类型/方法级注释。

---

### 总结

实现整体质量高：HMAC-SHA256 + crypto/rand 密钥 + hmac.Equal 常量时间比较 + HttpOnly/SameSite=Strict/条件 Secure。DI 接线符合项目模式，向后兼容完整（API key 认证保留为优先级 1/2）。P1 #1 是实际 correctness bug（约12%概率合法 cookie 被拒绝），建议修复合并。

File: internal/security/cookie_test.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+	time.Sleep(10 * time.Millisecond)
+
+	cookies := w.Result().Cookies()
+	require.Len(t, cookies, 1)
+
+	r2 := httptest.NewRequest("GET", "/", nil)
+	r2.AddCookie(cookies[0])
+
+	uid, ok := ca.Authenticate(r2)
+	require.False(t, ok, "expired cookie should be rejected")
```
Comment (hotplex-ai) [inline internal/security/cookie_test.go:68]: **[P2] `time.Sleep` 违反项目测试规范**（CLAUDE.md: 禁止 time.Sleep 等待）。`maxAge` 已设为 1ns，cookie 时间戳在 `sign()` 时已被签入，验证时 `time.Since(ts)` 必然超过 maxAge，无需等待。可直接移除此行。

**Review by hotplex-ai — DISMISSED** [review]
## Code Review — hotplex-ai

**Verdict: APPROVE** | P0:0 P1:0 P2:3 P3:1

PR#625 HMAC cookie auth 实现质量高：密码学实践正确（HMAC-SHA256 + crypto/rand + hmac.Equal 常量时间比较），cookie 安全属性完备（HttpOnly/SameSite=Strict/conditional Secure/24h 过期），向后兼容性完美（X-API-Key 和 query param 认证路径完全保留），测试覆盖全面（10 个测试含 WS 升级集成测试）。

### P2 Findings

**[P2] Shared "webchat_user" 身份缺少启动日志提醒** (`internal/webchat/server.go:80`) — 所有 webchat 访客共享 `"webchat_user"` 身份，TODO 已记录。建议在 `gateway_run.go` webchat 启用路径增加一条 warn 级别日志提醒运维者此限制，避免多用户场景下的静默授权混淆。

**[P2] 安全模型文档未更新** (`docs/explanation/security-model.md`) — 当前安全模型文档仅描述 API key + deferred auth 两条路径，缺少第三条 HMAC cookie auth 路径。建议补充 cookie 签发/验证/优先级描述。

**[P2] X-Forwarded-Proto 无条件信任可加防御性日志** (`internal/security/cookie.go:149-153`) — 注释已充分说明 self-DoS 而非安全绕过，但可在非 TLS 连接检测到此 header 时输出 warn 日志。

### P3 Findings

**[P3] 设计 spec 仍标记 Draft** — 实现已完成并合并，建议更新 spec 状态为 Implemented。

### Verified
- ✅ HMAC-SHA256 + crypto/rand 密钥生成
- ✅ hmac.Equal 常量时间比较（防 timing attack）
- ✅ Cookie 安全属性：HttpOnly + SameSite=Strict + conditional Secure
- ✅ 24h 过期 + timestamp 新鲜度检查
- ✅ 向后兼容：X-API-Key header / query param 完全保留
- ✅ 跨源部署（HOTPLEX_WEBCHAT_WS_URL）继续使用 API key
- ✅ CI 全绿（Build + Test + Coverage + Lint）

File: docs/superpowers/specs/2026-06-02-webchat-cookie-auth-design.md [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # Webchat Cookie 认证设计规格
```
Comment (hotplex-ai) [inline docs/superpowers/specs/2026-06-02-webchat-cookie-auth-design.md:1]: [P2] Spec shows `SetCookie(w, userID)` but implementation is `SetCookie(w, r, userID)`. Update to match the actual signature.

File: internal/gateway/hub_test.go (hunk 1 above)
Comment (hotplex-ai) [inline internal/gateway/hub_test.go:1041]: [P2] `TestHub_HandleHTTP_CookieAuth` and `TestHub_HandleHTTP_CookieAuth_InvalidCookie` create isolated state — should call `t.Parallel()` per project convention.

**Review by hotplex-ai — DISMISSED** [review]
## Code Review — hotplex-ai

Verdict: **APPROVE** | P0:0 P1:0 P2:3 P3:0

**Scope**: HMAC cookie auth for webchat — eliminate build-time API key exposure. 14 files (Go backend + TS frontend + design spec).

### Findings

**[P2] Design spec `SetCookie` signature not updated** (`docs/superpowers/specs/2026-06-02-webchat-cookie-auth-design.md:57`)
Spec shows `SetCookie(w, userID)` but implementation is `SetCookie(w, r, userID)`. The `r *http.Request` param was added for the no-op cookie optimization. Minor spec staleness — update signature to match implementation.

**[P2] Missing `t.Parallel()` in new hub cookie tests** (`internal/gateway/hub_test.go`)
`TestHub_HandleHTTP_CookieAuth` and `TestHub_HandleHTTP_CookieAuth_InvalidCookie` create isolated state but don't call `t.Parallel()`, inconsistent with project convention and the rest of the suite.

**[P2] Dual auth path asymmetry: WS fallback vs REST 401 on expired cookie** (`internal/gateway/hub.go:392-399` vs `internal/security/auth.go:68-74`)
WS upgrade falls through to `pendingAuth` on cookie failure, but REST `AuthenticateRequest` returns 401. A webchat user with expired cookie gets WS connected but REST calls fail. Consider having webchat client handle 401 by triggering a page reload to refresh the cookie.

### Positive Notes

- HMAC-SHA256 construction cryptographically sound with `hmac.Equal` constant-time comparison
- Cookie attributes correct: HttpOnly, SameSite=Strict, conditional Secure via `isHTTPS()`
- Secret via `crypto/rand`, never persisted to disk — eliminates build-time exposure
- `CookieAuth` immutability documented, lock-free concurrent access is correct
- Test coverage thorough (sign/verify, expiry, tamper, secure flag, repeat issue, authenticator integration)
- Cross-origin deployments continue via `X-API-Key` header fallback

File: internal/security/cookie.go [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+// sign creates a signed cookie value: Base64(timestamp|userID|hexHMAC).
+// The HMAC signature is hex-encoded to avoid binary bytes in the delimited format,
+// making the cookie value safe for debugging and unambiguous to parse.
+func (c *CookieAuth) sign(userID string) string {
+	ts := time.Now().Unix()
+	payload := fmt.Sprintf("%d|%s", ts, userID)
+
+	mac := hmac.New(sha256.New, c.secret)
+	mac.Write([]byte(payload))
+	sig := hex.EncodeToString(mac.Sum(nil))
```
Comment (hotplex-ai) [inline internal/security/cookie.go:96]: **[P2] userID `|` delimiter collision** — `sign()` uses `|` as delimiter: `ts|userID|hexHMAC`. If a future caller passes a userID containing `|`, `SplitN(raw, \"|\", 3)` in `verify()` breaks the roundtrip. Currently safe (`\"webchat_user\"` has no pipes), but the public `SetCookie` API accepts arbitrary strings. Consider: (a) validate userID rejects `|` in `SetCookie`, or (b) use length-prefixed encoding instead of delimiter-based parsing.

File: internal/security/cookie.go [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+//
+// Cookie format: Base64(timestamp|userID|hex(HMAC-SHA256(timestamp|userID, secret)))
+// Cookie attributes: HttpOnly, SameSite=Strict, Path=/, conditional Secure, 24h Max-Age.
+//
+// Immutability: the secret and maxAge fields are set once at construction and never
+// modified. This allows safe concurrent access from both Hub.HandleHTTP (WS upgrade)
+// and Authenticator.AuthenticateRequest (REST API) without additional locking.
+type CookieAuth struct {
+	secret []byte
+	maxAge time.Duration
```
Comment (hotplex-ai) [inline internal/security/cookie.go:39]: **[P2] Gateway restart invalidates all cookies** — HMAC secret is `crypto/rand` generated on every `NewCookieAuth()` call. After restart, all existing browser cookies silently fail and fall through to `pendingAuth`. Users need a page reload (SPA fallback re-issues). By design, but worth documenting in the design spec as an operational note.

File: internal/gateway/hub.go [hunk 9]
```diff
@@ -388,6 +389,14 @@ func (h *Hub) HandleHTTP(
 			}
 			userID = uid
 			botID = security.BotIDFromRequest(r)
+		} else if cookieAuth != nil {
```
Comment (hotplex-ai) [inline internal/gateway/hub.go:392]: **[P2] Dual cookie auth paths** — Cookie auth is implemented independently here (direct `cookieAuth.Authenticate`) and in `AuthenticateRequest` (via `a.cookieAuth`). The `cookieAuth` parameter is redundant since `auth` already holds it via `SetCookieAuth`. Future changes to one path could miss the other. Consider routing through `Authenticator` for a single code path.

File: internal/security/cookie.go [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+	return &CookieAuth{
+		secret: secret,
+		maxAge: cookieMaxAge,
+	}, nil
+}
+
+// SetCookie issues a new HMAC-signed cookie for the given userID.
+// If the request already carries a valid cookie, this is a no-op (avoids
+// resetting the expiry on every page refresh).
+func (c *CookieAuth) SetCookie(w http.ResponseWriter, r *http.Request, userID string) error {
```
Comment (hotplex-ai) [inline internal/security/cookie.go:57]: **[P3] `SetCookie` error always nil** — The `error` return is never non-nil and all callers discard it with `_ =`. Consider removing the return value or adding a comment documenting the contract for future callers.

**Review by hotplex-ai — COMMENTED** [review]
## Code Review — hotplex-ai

Verdict: ✅ APPROVE | P0:0 P1:0 P2:4 P3:4

### P2 Findings

1. **userID `|` delimiter collision in cookie format** (`internal/security/cookie.go:96-108`) — `sign()` formats `ts|userID|hexHMAC` and `verify()` uses `SplitN(raw, "|", 3)`. If a future caller passes a userID containing `|`, the roundtrip silently breaks. Currently safe (hardcoded `"webchat_user"`), but the public API accepts arbitrary strings. Consider adding validation in `SetCookie` or using a length-prefixed format.

2. **Gateway restart invalidates all webchat cookies** (`internal/security/cookie.go:37-40`) — HMAC secret is regenerated fresh on every startup. After restart, existing browser cookies fall through to `pendingAuth`; users need a page reload to re-establish cookie auth via SPA fallback. By design (ephemeral secret), but worth noting in the design spec as an operational consideration.

3. **Dual cookie auth code paths** (`internal/gateway/hub.go:392` vs `internal/security/auth.go:68`) — Cookie auth is implemented independently in `HandleHTTP` (direct `cookieAuth.Authenticate`) and `AuthenticateRequest` (via `a.cookieAuth`). The `HandleHTTP` parameter is redundant since `auth` already holds `cookieAuth` via `SetCookieAuth`. Future modifications to one path could miss the other. Consider routing through a single method on `Authenticator`.

4. **Missing test: userID with special characters** (`internal/security/cookie_test.go`) — Test suite covers normal/expired/tampered cookies but not userID edge cases (`|`, empty string, very long). A test with `userID="a|b"` would surface the SplitN issue above.

### P3 Findings

5. **`SetCookie` error return is always nil** (`internal/security/cookie.go:57`) — The `error` return is never non-nil and callers discard it with `_ =`. Consider removing it or documenting the contract.

6. **`isHTTPS` trusts X-Forwarded-Proto unconditionally** (`internal/security/cookie.go:149`) — Self-DoS risk if gateway is accessed directly without reverse proxy: attacker sets header to force `Secure` cookie over HTTP, blocking cookie auth. Comment correctly identifies this. Low priority.

7. **`isSameOrigin()` detection relies on env var absence** (`webchat/lib/config.ts:36`) — The heuristic `!process.env.HOTPLEX_WEBCHAT_WS_URL` works for embedded mode but could confuse operators who set `WS_URL` to the same origin. Worth documenting in webchat configuration docs.

8. **Invalid cookie test does not verify pendingAuth state** (`internal/gateway/hub_test.go:1233`) — Test confirms WS upgrade succeeds but does not assert the connection is actually in `pendingAuth` state. Strengthening this would improve regression coverage.

**Review by hotplex-ai — APPROVED** [review]
## Code Review — hotplex-ai (Follow-up)

Verdict: ✅ APPROVE | P0:0 P1:0 P2:4 P3:4

All findings documented in previous review comment. No blocking issues — safe to merge.
